sdk/confidentialledger/azure-confidentialledger/azure/confidentialledger/receiptverification/__init__.py

```diff
@@ -0,0 +1,18 @@
+# coding=utf-8
+# --------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------
+
+from .receipt_verification import verify_receipt, verify_receipt_from_dict
+from .models import Receipt, LeafComponents, ProofElement
+from ._version import VERSION
+
+__version__ = VERSION
+__all__ = [
+    "verify_receipt",
+    "verify_receipt_from_dict",
+    "Receipt",
+    "LeafComponents",
+    "ProofElement",
```

Comment:
How are these other objects used? I don't see them in the samples, so I'm not sure why these models are public.

Reply:
The idea is to offer a receipt verification method from either a JSON object (which is the receipt format received from the `GetReceipt` call) or from specific model objects that can be manually created by clients (e.g., if they want to manually craft a receipt and verify it for their own purpose). I only documented the verification from a dictionary in the sample / README as that seems the most common user scenario.

Do you think we should also document the verification from these models? Or, do you believe these models should not even be exposed and only be used internally? I see some advantage in exposing them, but if there's a strong reason to not have them public, I can definitely remove them.

Reply:
I'd turn it around and say that we generally don't expose models *unless* we have a strong motivation to do so. All public API surface area comes with a cost - both from a test and documentation perspective as well as it severely limits what changes/improvements we can do to the APIs moving forward. It also increases the cognitive load on users - the more entries I see in statement completion/ToC in docs etc, the harder it gets to find what you are looking for.

Reply:
I see that makes sense. I made the models internal only and they are not exposed to customers anymore.